packages/core/test/financial-templates/perpetual-multiparty/PositionManager.js

```diff
@@ -0,0 +1,1503 @@
+    await collateral.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    const numTokens = toWei("100");
+    const amountCollateral = toWei("150");
+    await positionManager.create({ rawValue: amountCollateral }, { rawValue: numTokens }, { from: sponsor });
+
+    // Transfer half the tokens from the sponsor to a tokenHolder. IRL this happens through the sponsor selling tokens.
+    const tokenHolderTokens = toWei("50");
+    await tokenCurrency.transfer(tokenHolder, tokenHolderTokens, { from: sponsor });
+
+    // Some time passes and the UMA token holders decide that Emergency shutdown needs to occur.
+    const shutdownTimestamp = Number(await positionManager.getCurrentTime()) + 1000;
+    await positionManager.setCurrentTime(shutdownTimestamp);
+
+    // Should revert if emergency shutdown initialized by non-FinancialContractsAdmin (governor).
+    assert(await didContractThrow(positionManager.emergencyShutdown({ from: other })));
+
+    // FinancialContractAdmin can initiate emergency shutdown.
+    await financialContractsAdmin.callEmergencyShutdown(positionManager.address);
+    assert.equal(await positionManager.emergencyShutdownTimestamp(), shutdownTimestamp);
+    assert.equal((await positionManager.emergencyShutdownPrice()).toString(), 0);
+
+    // Because the emergency shutdown is called by the `financialContractsAdmin`, listening for events can not
+    // happen in the standard way as done in other tests. However, we can directly query the `positionManager`
+    // to see it's past events to ensure that the right parameters were emmited.
+    const eventResult = await positionManager.getPastEvents("EmergencyShutdown");
+    assert.equal(eventResult[0].args.caller, financialContractsAdmin.address);
+    assert.equal(eventResult[0].args.shutdownTimestamp.toString(), shutdownTimestamp.toString());
+
+    // Emergency shutdown should not be able to be called a second time.
+    assert(await didContractThrow(financialContractsAdmin.callEmergencyShutdown(positionManager.address)));
+
+    // Before the DVM has resolved a price withdrawals should be disabled (as with settlement at maturity).
+    assert(await didContractThrow(positionManager.settleEmergencyShutdown({ from: sponsor })));
+
+    // All contract functions should also blocked as emergency shutdown.
+    assert(
+      await didContractThrow(
+        positionManager.create({ rawValue: toWei("1") }, { rawValue: toWei("1") }, { from: sponsor })
+      )
+    );
+    assert(await didContractThrow(positionManager.deposit({ rawValue: toWei("1") }, { from: sponsor })));
+    assert(await didContractThrow(positionManager.withdraw({ rawValue: toWei("1") }, { from: sponsor })));
+    assert(await didContractThrow(positionManager.redeem({ rawValue: toWei("1") }, { from: sponsor })));
+    assert(await didContractThrow(positionManager.requestWithdrawal({ rawValue: toWei("1") }, { from: sponsor })));
```

Comment:
Should also add in 
- `requestTransfer`
- `transferPassedRequest`
- `withdrawPassedRequest`
- `remargin`

Reply:
sure & done